tardis/grid/base.py

```diff
@@ -0,0 +1,172 @@
+import pandas as pd
+import numpy as np
+import copy
+import tardis
+
+from tardis.io.config_reader import Configuration
+from tardis.model import Radial1DModel
+
+
+def _set_tardis_config_property(tardis_config, key, value):
+    """
+    Sets tardis_config[key] = value for a TARDIS config
+    dictionary. Recursively searches for the deepest
+    dictionary and sets the corresponding value.
+
+    Parameters
+    ----------
+    tardis_config : tardis.io.config_reader.Configuration
+        The config where the key,value pair should be set.
+    key : str
+        The key in the tardis_config. Should be of the format
+        'property.property.property'
+    value : object
+        The value to assign to the config dict for the
+        corresponding key.
+    """
+    keyitems = key.split(".")
+    tmp_dict = getattr(tardis_config, keyitems[0])
+    for key in keyitems[1:-1]:
+        tmp_dict = getattr(tmp_dict, key)
+    setattr(tmp_dict, keyitems[-1], value)
+    return
+
+
+class tardisGrid:
+    """
+    A class that stores a grid of TARDIS parameters and
+    facilitates running large numbers of simulations
+    easily.
+
+    Parameters
+    ----------
+    configFile : str
+        path to TARDIS yml file.
+    gridFrame : pandas.core.frame.DataFrame
```

Comment:
Possibly do some type-checking here?  Could get weird if someone accidentally passes in a numpy array or a list

Reply:
I don't think it's necessary since there's effectively duck typing since the grid relies on `.iloc[]` for accessing rows.